WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -1873,16 +1959,143 @@ static GSourceFuncs _handlerIntervention =
             return URL.substr(beginDomain, endDomain - beginDomain);
         }
 
-        uint32_t URL(const string& URL) override
+        void AppendClientCertificate(const std::string& newWpeHost, const std::string& newWpeClientCert, const std::string& newWpeClientCertKey, std::string& newCertContents)
+        {
+            if (newWpeHost.empty() || newWpeClientCert.empty() || newWpeClientCertKey.empty()) {
+                newCertContents.clear();
+                TRACE(Trace::Error, (_T("Invalid arguments (%d,%d,%d)\n"), newWpeHost.empty(), newWpeClientCert.empty(), newWpeClientCertKey.empty()));
+                return;
+            }
+
+            OpenSSL_add_all_algorithms();
+
+            auto *certsUrls = getenv(ENV_WPE_CLIENT_CERTIFICATES_URLS);
+            if (certsUrls && *certsUrls)
+            {
+                constexpr auto kUrlsSep = ENV_URL_LIST_DELIMITER; // not allowed in URI
+                std::string urlsList(certsUrls);
+                auto urls = tokenize(urlsList, kUrlsSep);
+                auto append_newHost = true;
+                for (const auto &url : urls)
+                {
+                    if (url == newWpeHost)
+                    {
+                        TRACE(Trace::Information, (_T("Already found on the list %s\n "), newWpeHost.c_str()));
+                        append_newHost = false;
+                        break;
+                    }
+                }
+
+                if (append_newHost)
```

Comment:
I would append the host only if all the operations succeeded (certContents and keyContents are fine).

Reply:
Done